lib/pbench/agent/tool_meister.py

```diff
@@ -1285,6 +1285,50 @@ def stop_tools(self, data):
             self._send_client_status("success")
         return failures
 
+    def _create_tar(
+        self,
+        target_dir: Path,
+        tar_file: Path,
+        cwd: Path = None,
+        ignore_warning: bool = False,
+    ) -> subprocess.CompletedProcess:
+        """
+        Creates a tar file at a given tar file path.
+        """
+        tar_args = [
+            self.tar_path,
+            "--create",
+            "--xz",
+            "--force-local",
+            f"--file={tar_file}",
+            target_dir,
+        ]
+        if ignore_warning:
+            tar_args.insert(2, "--warning=none")
+
+        try:
+            # Invoke tar directly for efficiency.
+            cp = subprocess.run(
+                tar_args,
+                cwd=cwd if cwd else target_dir.parent,
+                stdin=None,
+                stdout=subprocess.PIPE,
+                stderr=subprocess.STDOUT,
+            )
+            if cp.returncode != 0:
+                self.logger.warning(
+                    "Failed to create tar ball with arguments %s; Error: %s",
+                    str(tar_args),
```

Comment:
Elsewhere, instead of stringifying a list of strings, we've been `join`'ing them to good effect:
```
                    " ".join(tar_args),
```